assembly/src/parsers/stack_ops.rs

```diff
@@ -162,52 +162,97 @@ pub fn parse_swap(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Assem
                 span_ops.push(Operation::MovDn7);
                 // MovUp8
                 span_ops.push(Operation::Pad);
-                span_ops.push(Operation::MovUp9);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::Swap);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::MovUp8);
                 span_ops.push(Operation::Add);
             }
             "9" => {
-                span_ops.push(Operation::MovDn9);
+                span_ops.push(Operation::MovDn8);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::Swap);
+                span_ops.push(Operation::SwapDW);
                 span_ops.push(Operation::Pad);
-                span_ops.push(Operation::MovUp9);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::Swap);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::MovUp8);
                 span_ops.push(Operation::Add);
             }
             "10" => {
-                span_ops.push(Operation::MovDn9);
+                span_ops.push(Operation::MovDn8);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::Swap);
+                span_ops.push(Operation::SwapDW);
                 // MovUp10
                 span_ops.push(Operation::Pad);
-                span_ops.push(Operation::MovUp11);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::MovUp3);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::MovUp8);
                 span_ops.push(Operation::Add);
             }
             "11" => {
-                span_ops.push(Operation::MovDn11);
+                span_ops.push(Operation::MovDn8);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::MovDn3);
+                span_ops.push(Operation::SwapDW);
                 span_ops.push(Operation::Pad);
-                span_ops.push(Operation::MovUp11);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::MovUp3);
+                span_ops.push(Operation::SwapDW);
+                span_ops.push(Operation::MovUp8);
                 span_ops.push(Operation::Add);
```

Comment:
I think we can do these a bit more efficiently. For example, for `swap.11` something like this should work:
```
swapdw
movup8
movdn4
movup3
movdn8
swapdw
```
If the above works, we can do the swap in 6 cycles rather than 10.